include/arch/arc/arch.h

```diff
@@ -89,55 +89,66 @@ extern "C" {
 		1 << (31 - __builtin_clz(x) + 1) : \
 		1 << (31 - __builtin_clz(x)))
 
-
-#if defined(CONFIG_USERSPACE)
-#define ARCH_THREAD_STACK_RESERVED \
-	(STACK_GUARD_SIZE + CONFIG_PRIVILEGED_STACK_SIZE)
-#else
-#define ARCH_THREAD_STACK_RESERVED (STACK_GUARD_SIZE)
-#endif
-
-
-#if  defined(CONFIG_USERSPACE) && CONFIG_ARC_MPU_VER == 2
+#if CONFIG_ARC_MPU_VER == 2
 /* MPUv2 requires
  *  - region size must be power of 2 and >= 2048
  *  - region start must be aligned to its size
  */
-#define Z_ARC_MPUV2_SIZE_ALIGN(size) POW2_CEIL(STACK_SIZE_ALIGN(size))
+#define Z_ARC_MPU_SIZE_ALIGN(size) POW2_CEIL(STACK_SIZE_ALIGN(size))
+#elif CONFIG_ARC_MPU_VER == 3
+/* MPUv3 requires
+ * - region size must be 32 bytes aligned
+ * - region start must be 32 bytes aligned
+ */
+#define Z_ARC_MPU_SIZE_ALIGN(size) STACK_SIZE_ALIGN(size)
```

Comment:
For maintainability, please explain that STACK_SIZE_ALIGN takes care of the 32 byte alignement - or, simply put the math that can be checked/understood here directly.

Reply:
side note: I've rewritten and standardized this across in my in-progress PR #24714. We have new macros: 

ARCH_STACK_PTR_ALIGN - Required alignment of the stack pointer
ARCH_THREAD_STACK_OBJ_ALIGN(size) - Required alignment of the base address of the stack object
ARCH_THREAD_STACK_SIZE_ADJUST(size) - Round up a requested stack size to meet constraints (MPU granularity, page boundary, reserved data, etc.)

STACK_ALIGN / STACK_SIZE_ALIGN are completely removed. They were improperly namespaced and didn't have the same semantics everywhere.